src/com/amazon/ion/benchmark/ParseAndCompareBenchmarkResults.java

```diff
@@ -0,0 +1,172 @@
+     * Extract the parameter of specific aspect from set (input | options) out of benchmark result.
+     * @param benchmarkResultFilePath is the path of benchmark result.
+     * @param keyWord specifies which parameter will be returned, from set (input | options).
+     * @return an Ion Value represents the parameter conforms with the keyword.
+     * @throws Exception if error occurs when reading Ion data.
+     */
+    private static IonValue getParameter(String benchmarkResultFilePath, String keyWord) throws Exception{
+        IonStruct benchmarkResultStruct = readHelper(benchmarkResultFilePath);
+        IonStruct parameterStruct = (IonStruct)benchmarkResultStruct.get(PARAMETERS);
+        return parameterStruct.get(keyWord);
+    }
+
+    /**
+     * This is a helper method which create a IonReader for benchmark result and extract the IonStruct which contain parameters or scores information.
+     * @param benchmarkResultFilePath is the path of benchmark result.
+     * @return an Ion Struct which contains information of parameters using during the benchmark process or scores.
+     * @throws Exception if error occurs when reading Ion Data.
+     */
+    private static IonStruct readHelper(String benchmarkResultFilePath) throws Exception {
+        try (IonReader reader = IonReaderBuilder.standard().build(new BufferedInputStream(new FileInputStream(benchmarkResultFilePath)))) {
+            reader.next();
+            reader.stepIn();
+            reader.next();
+            IonDatagram benchmarkResult = ReadGeneralConstraints.LOADER.load(reader);
+            return (IonStruct) benchmarkResult.get(0);
+        }
+    }
+
+    /**
+     * Calculate the relative difference between scores from benchmark results of different in-java commits.
+     * @param previousScore is score from the benchmark result of the existing ion-java commit.
+     * @param newScore is score from the benchmark result of the new ion-java commit.
+     * @return relative changes of two scores from different benchmark results in BigDecimal format.
+     */
+    private static BigDecimal calculateDifference(BigDecimal previousScore, BigDecimal newScore) {
+        BigDecimal scoreDifference = newScore.subtract(previousScore);
+        return scoreDifference.divide(previousScore, RoundingMode.HALF_UP);
+    }
+
+    /**
+     * Write calculated relative changes of scores in an Ion Struct into the generated file.
+     * @param benchmarkResult is the path of benchmark result.
+     * @param outputFilePath is destination path of generated result.
+     * @param scoreMap is a hashmap which match score with the aspect it represents.
+     * @throws Exception if error occurs when reading Ion data.
+     */
+    private static void writeResult(String benchmarkResult, String outputFilePath, Map<String, BigDecimal> scoreMap) throws Exception {
+        File file = new File(outputFilePath);
+        String inputFileName = getParameter(benchmarkResult, INPUT).toString();
```

Comment:
I believe if you cast this to `IonString` and use `stringValue()` that will give you the raw string value, so you won't need to use `inputFileName.replace("\"", "")` later.

Reply:
> I believe if you cast this to `IonString` and use `stringValue()` that will give you the raw string value, so you won't need to use `inputFileName.replace("\"", "")` later.

That worked, thanks for suggestion, changes will be updated in the next commit